### PR TITLE
renamed user context to visitor target group

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | depends on #3335 
| License | MIT
| Documentation PR | ---

#### What's in this PR?

We've decided to rename the user context to visitor target group, because the context is not a hash as in the FOSHttpCacheBundle anymore.

### CODE REVIEW

File: src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/EventListener/TargetGroupSubscriberTest.php [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+
+        $this->assertEquals('<body><script></script></body>', $response->getContent());
+    }
+
+    public function provideAddTargetGroupHitScript()
+    {
+        return [
+            ['/_target_group_hit', 'X-Forwarded-URL', 'X-Fowarded-Referer', 'X-Forwarded-UUID', 'some-uuid'],
+            ['/_target_group_hit', 'X-Forwarded-URL', 'X-Fowarded-Referer', 'X-Forwarded-UUID', null],
+            ['/context_hit', 'X-Other-URL', 'X-Other-Referer', 'X-Uuid', 'some-other-uuid'],
```
Comment (chirimoya) [inline src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/EventListener/TargetGroupSubscriberTest.php]: `context_hit` ???

File: src/Sulu/Bundle/AudienceTargetingBundle/TargetGroup/TargetGroupStore.php [hunk 2]
```diff
@@ -9,17 +9,17 @@
  * with this source code in the file LICENSE.
  */
 
-namespace Sulu\Bundle\AudienceTargetingBundle\UserContext;
+namespace Sulu\Bundle\AudienceTargetingBundle\TargetGroup;
 
 /**
  * This class stores the current user context. Also allows updating it, in which case it is marked as changed.
```
Comment (chirimoya) [inline src/Sulu/Bundle/AudienceTargetingBundle/TargetGroup/TargetGroupStore.php]: user context

File: src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/EventListener/TargetGroupSubscriberTest.php [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+        $this->targetGroupStore->updateTargetGroupId(Argument::any())->shouldNotBeCalled();
+
+        $targetGroupSubscriber->setTargetGroup($event->reveal());
+    }
+
+    public function provideSetTargetGroupFromHeader()
+    {
+        return [
+            ['X-Sulu-Target-Group', '1', '1'],
+            ['X-Context', '2', '2'],
```
Comment (chirimoya) [inline src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/EventListener/TargetGroupSubscriberTest.php]: `X-Context` ???

File: src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/EventListener/TargetGroupSubscriberTest.php [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+        }
+
+        $targetGroupSubscriber->setTargetGroup($event->reveal());
+    }
+
+    public function provideSetTargetGroupFromCookie()
+    {
+        return [
+            ['sulu-visitor-target-group', 'visitor-session', '1', true, null, '1', false],
+            ['context', 'context-session', '3', true, null, '3', false],
```
Comment (chirimoya) [inline src/Sulu/Bundle/AudienceTargetingBundle/Tests/Unit/EventListener/TargetGroupSubscriberTest.php]: `context` ???

**Comment by chirimoya** [comment]
@danrot ... documentation update is missing.

**Comment by danrot** [comment]
I was waiting for your feedback before changing the documentation. Here it is: https://github.com/sulu/sulu-docs/pull/325

**Pull request merged**